extensions/blocks/simple-payments/simple-payments.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Pay with PayPal block (aka Simple Payments).
+ *
+ * @since 9.0.0
+ *
+ * @package Jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\SimplePayments;
+
+use Jetpack_Simple_Payments;
+
+const FEATURE_NAME = 'simple-payments';
+const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
+
+/**
+ * Registers the block for use in Gutenberg
+ * This is done via an action so that we can disable
+ * registration if we need to.
+ */
+function register_block() {
+	jetpack_register_block(
+		BLOCK_NAME,
+		array(
+			'render_callback' => __NAMESPACE__ . '\render_block',
+			'plan_check'      => true,
+		)
+	);
+}
+add_action( 'init', __NAMESPACE__ . '\register_block' );
+
+/**
+ * Pay with PayPal block dynamic rendering.
+ *
+ * @param array  $attr    Array containing the block attributes.
+ * @param string $content String containing the block content.
+ *
+ * @return string
+ */
+function render_block( $attr, $content ) {
+	// Do nothing if block content is a `simple-payment` shortcode.
+	if ( preg_match( '/\[simple-payment(.*)]/', $content ) ) {
+		return $content;
+	}
+
+	$simple_payments = Jetpack_Simple_Payments::getInstance();
+	$simple_payments->enqueue_frontend_assets();
+
+	// Keep content as-is if rendered in other contexts than frontend (i.e. feed, emails, API, etc.).
+	if ( ! jetpack_is_frontend() ) {
+		return $content;
+	}
```

Comment:
Should this be before we enqueue assets, since we won't have them in contexts outside of the WP frontend?

Reply:
https://github.com/Automattic/jetpack/pull/17256/commits/6831acecb379699d2938ad8a9e2a1f48128ded13